contrib/checkbox-provider-ce-oem/bin/parse_optee_test.py

```diff
@@ -2,32 +2,69 @@
 
 import json
 import argparse
+import subprocess as sp
+import shlex
+import re
+from checkbox_support.snap_utils.snapd import Snapd
+from checkbox_support.snap_utils.system import get_gadget_snap
 
 
-def parse_json_file(filename, filter=False):
-    with open(filename, 'r') as file:
+def parse_json_file(filepath, filter=False, xtest=None):
+    with open(filepath, 'r') as file:
         data = json.load(file)
     for test in data:
         if (filter and test['suite'] == 'pkcs11') or \
            (not filter and test['suite'] not in ['pkcs11']):
-            print_test_info(test)
+            print_test_info(test, xtest)
 
 
-def print_test_info(test):
+def print_test_info(test, xtest):
     print("suite: {}".format(test['suite']))
     print("test_id: {}".format(test['test_id']))
-    print("description: {}\n".format(test['test_description']))
+    print("description: {}".format(test['test_description']))
+    print("tool: {}\n".format(xtest))
+
+
+def look_up_xtest():
+    xtest = look_up_gadget()
+    if xtest is not False:
+        pass
+    elif Snapd().list("x-test"):
+        xtest = "x-test.xtest"
+    else:
+        xtest = None
+    return xtest
```

Comment:
Adjust the order, if `x-test` snap exists, use it, else fallback to `gadget`. What do you think?

Reply:
@baconYao I've updated the logic to prioritize the x-test snap. Thanks for the review.